webapp/channels/src/components/admin_console/cluster_settings.tsx

```diff
@@ -3,36 +3,43 @@
 
 import React from 'react';
 import {FormattedMessage} from 'react-intl';
+import {AdminConfig, ClientLicense, ClusterSettings as ClusterSettingsType} from '@mattermost/types/config';
 
 import {Client4} from 'mattermost-redux/client';
 
 import * as Utils from 'utils/utils';
 
 import WarningIcon from 'components/widgets/icons/fa_warning_icon';
+import {DeepPartial} from '@mattermost/types/utilities';
 
 import ExternalLink from 'components/external_link';
 
-import AdminSettings from './admin_settings';
+import AdminSettings, {BaseProps, BaseState} from './admin_settings';
 import BooleanSetting from './boolean_setting';
 import ClusterTableContainer from './cluster_table_container.jsx';
 import SettingsGroup from './settings_group';
 import TextSetting from './text_setting';
 
-export default class ClusterSettings extends AdminSettings {
-    getConfigFromState = (config) => {
+export type ClusterSettingsProps = BaseProps & {
+    license: ClientLicense;
+};
+type State = BaseState & ClusterSettingsType;
```

Comment:
This component states although they appear to match that of CLusterSettingsType. Might not always be true. For example	`BindAddress` is missing state in this component where as its present in the type. can you rewrite this State based on this component.